chrome/content/zotero/xpcom/sync/syncAPIClient.js

```diff
@@ -30,12 +30,16 @@ if (!Zotero.Sync) {
 Zotero.Sync.APIClient = function (options) {
 	if (!options.baseURL) throw new Error("baseURL not set");
 	if (!options.apiVersion) throw new Error("apiVersion not set");
-	if (!options.apiKey) throw new Error("apiKey not set");
+	if (!(options.apiKey || (options.username && options.password))) {
+		throw new Error("neither apiKey nor (username, password) is set");
+	}
 	if (!options.caller) throw new Error("caller not set");
 	
 	this.baseURL = options.baseURL;
 	this.apiVersion = options.apiVersion;
 	this.apiKey = options.apiKey;
+	this.username = options.username;
+	this.password = options.password;
```

Comment:
This is a matter of creating the `APIClient` without passing in `apiKey` (which throws an exception), so I figured I may as well store the credentials. In general was not certain what's the best way to instantiate `APIClient` without the API key. Maybe just pass in an empty string instead in `Sync.Runner` or set up a custom flag like `noApiKey`?